src/java/org/apache/cassandra/schema/Types.java

```diff
@@ -498,16 +570,101 @@ public long serializedSize(Types t, Version version)
+                List<FieldIdentifier> fieldIdentifiers = type.fieldNames();
+                List<String> fieldNames = fieldIdentifiers.stream().map(FieldIdentifier::toString).collect(toList());
+                List<String> fieldTypes = type.fieldTypes().stream().map(AbstractType::asCQL3Type).map(CQL3Type::toString).collect(toList());;
                 size += sizeof(fieldNames.size());
                 for (String s : fieldNames)
                     size += sizeof(s);
                 size += sizeof(fieldTypes.size());
                 for (String s : fieldTypes)
                     size += sizeof(s);
+                if (version.isAtLeast(Version.V8))
+                {
+                    size += sizeof(type.comment);
+                    size += sizeof(type.securityLabel);
+                    size += fieldMetadataSize(type, fieldIdentifiers);
+                }
             }
             return size;
         }
+
+        private void serializeFieldMetadata(UserType type, List<FieldIdentifier> fieldIdentifiers, DataOutputPlus out) throws IOException
+        {
+            // Sparse serialization: only serialize non-empty metadata with their positions
+            serializeSparseMetadata(fieldIdentifiers, type::fieldComment, out);
+            serializeSparseMetadata(fieldIdentifiers, type::fieldSecurityLabel, out);
+        }
+
+        private void serializeSparseMetadata(List<FieldIdentifier> fieldIdentifiers,
+                                            java.util.function.Function<FieldIdentifier, String> metadataGetter,
+                                            DataOutputPlus out) throws IOException
+        {
+            // Count non-empty values
+            int nonEmptyCount = 0;
+            for (FieldIdentifier fieldId : fieldIdentifiers)
+            {
+                if (!metadataGetter.apply(fieldId).isEmpty())
+                    nonEmptyCount++;
+            }
+
+            // Write count followed by position-value pairs
+            out.writeInt(nonEmptyCount);
+            for (int i = 0; i < fieldIdentifiers.size(); i++)
+            {
+                String value = metadataGetter.apply(fieldIdentifiers.get(i));
+                if (!value.isEmpty())
+                {
+                    out.writeInt(i);
+                    out.writeUTF(value);
+                }
+            }
+        }
+
+        private void deserializeFieldMetadata(DataInputPlus in, List<String> fieldComments, List<String> fieldSecurityLabels) throws IOException
+        {
+            deserializeSparseMetadata(in, fieldComments);
+            deserializeSparseMetadata(in, fieldSecurityLabels);
+        }
+
+        private void deserializeSparseMetadata(DataInputPlus in, List<String> target) throws IOException
+        {
+            int count = in.readInt();
+            for (int i = 0; i < count; i++)
+            {
+                int position = in.readInt();
```

Comment:
Update here too, if you have updated the serialization method. 

```suggestion
                int position = in.readUnsignedVInt32();
```